From: Jiri Slaby <jslaby@suse.cz>
Date: Fri, 12 Feb 2016 10:53:24 +0100
Subject: Revert "ipv6: tcp: add rcu locking in tcp_v6_send_synack()"
Patch-mainline: no, kabi
References: bnc#961257

This reverts commit ea28b18b992f25d05632725c33ff79b943b6d02f, upstream
commit 3e4006f0b86a5ae5eb0e8215f9a9e1db24506977. It is dependent on
the commit reverted in:
patches.kabi/revert-ipv6-add-complete-rcu-protection-around-np-op.patch

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 net/ipv6/tcp_ipv6.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/net/ipv6/tcp_ipv6.c b/net/ipv6/tcp_ipv6.c
index 7138ee87e07c..90004c6e3bff 100644
--- a/net/ipv6/tcp_ipv6.c
+++ b/net/ipv6/tcp_ipv6.c
@@ -484,10 +484,8 @@ static int tcp_v6_send_synack(struct sock *sk, struct dst_entry *dst,
 
 		fl6->daddr = treq->rmt_addr;
 		skb_set_queue_mapping(skb, queue_mapping);
-		rcu_read_lock();
 		err = ip6_xmit(sk, skb, fl6, rcu_dereference(np->opt),
 			       np->tclass);
-		rcu_read_unlock();
 		err = net_xmit_eval(err);
 	}
 
-- 
2.7.1

